addons/product_expiry/models/production_lot.py

```diff
@@ -34,21 +34,38 @@ def _get_dates(self, product_id=None):
         res = dict.fromkeys(mapped_fields.keys(), False)
         product = self.env['product.product'].browse(product_id) or self.product_id
         if product:
+            op_date = datetime.datetime.now()
+            if life_date:
+                res['life_date'] = fields.Datetime.from_string(life_date)
+                op_date = res['life_date'] - datetime.timedelta(days=product.life_time)
+
             for field in mapped_fields.keys():
                 duration = getattr(product, mapped_fields[field])
                 if duration:
-                    date = datetime.datetime.now() + datetime.timedelta(days=duration)
+                    date = op_date + datetime.timedelta(days=duration)
                     res[field] = fields.Datetime.to_string(date)
         return res
 
     # Assign dates according to products data
     @api.model
     def create(self, vals):
-        vals.update(self._get_dates(vals.get('product_id')))
+        vals.update(self._get_dates(vals.get('product_id'), vals.get('life_date')))
```

Comment:
The method get() returns a value for the given key. If key is not available then returns default value None.